controllers/metal3.io/host_config_data.go

```diff
@@ -94,6 +94,31 @@ func (hcd *hostConfigData) NetworkData() (string, error) {
 	return networkDataRaw, err
 }
 
+// PreprovisioningNetworkData get preprovisioning network configuration
+func (hcd *hostConfigData) PreprovisioningNetworkData() (string, error) {
+	if hcd.host.Spec.PreprovisioningNetworkDataName == "" {
+		hcd.log.Info("PreprovisioningNetworkDataName is not set, returning empty data")
+		return "", nil
+	}
+	networkData := &corev1.SecretReference{
```

Comment:
nit: it looks like you don't use the SecretReference itself, only its fields?

Reply:
Ah good point thanks - it's a copy/paste from NetworkData above which does have to handle SecretReference, will remove!